src/ompl/geometric/planners/experience/LightningRetrieveRepair.h

```diff
@@ -0,0 +1,202 @@
+
+            /**
+             * \brief Pass a pointer of the database from the lightning framework
+             */
+            void setLightningDB(ompl::tools::LightningDBPtr &experienceDB);
+
+            /** \brief Set the planner that will be used for repairing invalid paths recalled from experience */
+            void setRepairPlanner(const base::PlannerPtr &planner);
+
+            /** \brief See base class */
+            virtual void setup(void);
+
+            /**
+             * \brief Repairs a path to be valid in the current planning environment
+             * \param path - old from experience
+             * \param ptc - when to stop attempting repair
+             * \return true if no error
+             */
+            bool repairPath(ompl::geometric::PathGeometric &path, const base::PlannerTerminationCondition &ptc);
+
+            /**
+             * \brief Use our secondary planner to find a valid path between start and goal, and return that path
+             * \param start
+             * \param goal
+             * \param newPathSegment - the solution
+             * \return true if path found
+             */
+            bool replan(const ompl::base::State *start, const ompl::base::State *goal, ompl::geometric::PathGeometric &newPathSegment,
+                        const base::PlannerTerminationCondition &ptc);
+          
+            /**
+             * \brief Getter for number of 'k' close solutions to choose from database for further filtering 
+             */ 
+            int getNumNearestSolutions() const
+            {
+              return nearestK_;
+            }
+
+            /**
+             * \brief Setter for number of 'k' close solutions to choose from database for further filtering
+             */
+            void setNumNearestSolutions(int nearestK)
+            {
+              nearestK_ = nearestK;
+            }
+                    
+        protected:
+
+            /**
+             * \brief Count the number of states along the discretized path that are in collision
+             *        Note: This is kind of an ill-defined score though. It depends on the resolution of collision checking. 
+             *        I am more inclined to try to compute the percent of the length of the motion that is valid. 
+             *        That could go in SpaceInformation, as a utility function.
+             */
+            std::size_t checkMotionScore(const ompl::base::State *s1, const ompl::base::State *s2) const;
+
+            /**
+             * \brief Filters the top n paths in nearestPaths_ to the top 1, based on state validity with current environment
+             * \return true if no error
+             */
+            bool findBestPath(const base::State *startState, const base::State *goalState, ompl::base::PlannerDataPtr& chosenPath);
+
+            /** \brief The database of motions to search through */
+            ompl::tools::LightningDBPtr                           experienceDB_;
```

Comment:
i removed them throughout all the code, though I'm not sure I agree that is very user friendly since others can't just copy and paste the code into theirs